src/kubeobjects/secret.go

```diff
@@ -46,16 +46,29 @@ func (query SecretQuery) Update(secret corev1.Secret) error {
 	return errors.WithStack(query.kubeClient.Update(query.ctx, &secret))
 }
 
-func (query SecretQuery) CreateOrUpdate(secret corev1.Secret) error {
-	err := query.Create(secret)
-
-	if !k8serrors.IsAlreadyExists(err) {
-		return errors.WithStack(err)
+func (query SecretQuery) CreateOrUpdate(secret corev1.Secret) (bool, error) {
```

Comment:
nitpick: why is the `bool` return needed ?

if the update failed ==> we get an error
if no update happened ==> we don't care (only in tests)

does that worth having it ? 
maybe we could use the `creationTimestamp` in tests to check if it updated ? (if it gets updated)

Reply:
we need an actual return value for the `Reconcile()` method of the `Reconciler` to know if something changed